lib/grammar.pegjs

```diff
@@ -503,21 +538,44 @@ array_value_list = first:value _ rest:(',' _ value _)* {
     return [first].concat(take(rest, 2));
 }
 
+arg_map_value = 'makeArgMap' _ '(' _ first:( ident _ ':' _ type_ref ) _ rest:(',' _ ident _ ':' _ type_ref _ )* _ ')' {
+    let map = {};
+    map[first[0]] = first[4];
+    if (rest.length)
+        map[take(rest, 2)] = take(rest, 6);
+    return Ast.Value.ArgMap(map);
+}
+
+
+annotation = nl_annotation / impl_annotation
+nl_annotation = '#' '_' '[' name:ident _ '=' value:value ']' {
+    return { type: 'nl', name: name, value: value };
+}
+impl_annotation = '#' '[' name:ident _ '=' value:value ']' {
+    return { type: 'impl', name: name, value: value };
+}
+
 // Types
 
-type_ref = 'Measure' _ '(' _ unit:ident? _ ')' { return Type.Measure(unit); } /
+type_ref = hashable_type_ref /
```

Comment:
do you still use this distinction?

Reply:
if we don't need it, it's just extra cognitive load on our part when we read this code later on; as in, we'll come back 3 months from now to this code and wonder: "what was this again?" "is my new type hashable or not?"